libs/s3/s3_versioning_common_test_lib.py

```diff
@@ -344,9 +342,14 @@ def upload_versions(s3_test_obj: S3TestLib, s3_ver_test_obj: S3VersioningTestLib
             ...
             }
     """
-    LOG.info("Creating bucket")
-    resp = s3_test_obj.create_bucket(bucket_name)
-    assert_utils.assert_true(resp[0], resp[1])
+    try:
+        bucket_exists, _ = s3_test_obj.head_bucket(bucket_name)
+        if bucket_exists:
+            LOG.info("Bucket exists: %s, skipping bucket creation", bucket_name)
+    except CTException:
+        LOG.info("Creating bucket: %s", bucket_name)
+        resp = s3_test_obj.create_bucket(bucket_name)
+        assert_utils.assert_true(resp[0], resp[1])
```

Comment:
Keeping the code flexible here, in case we need to create buckets not created in test setup already. Aim is to keep test code a little leaner wherever possible.